lib/new_relic/agent.rb

```diff
@@ -293,6 +294,46 @@ def notice_error(exception, options = {})
       nil # don't return a noticed error data structure. it can only hurt.
     end
 
+    # Set a callback proc for determining an error's error group name
```

Comment:
Potential refactoring opportunity, though idk how it might impact Rdoc: 

This file is already very long. I think we've been adding things so that the APIs can all be accessed in the same way, but we could still do this if each API was in its own module that was included by NewRelic::Agent.

```ruby
module Child
	def child_method
		puts 'hi data!'
	end
end

module Parent
	extend self 
	include Child
end 

Parent.child_method 
# hi data!
# => nil                                                  
```

Reply:
We have at least these concerns currently with `lib/new_relic/agent.rb`:

- The customer facing RDoc incorrectly lists class methods as instance ones. For example, `NewRelic::Agent.notice_error` gets listed as `NewRelic::Agent#notice_error`.
- There are multiple classes being defined in the file
- The file has grown a bit unwieldy for devs to maintain, and could potentially benefit from being broken up a bit

There is, however, a big benefit that I can see for keeping all of the public methods inside the file, and that's that it will make them easier to find for customers. If you are told about a public API method, `NewRelic::Agent.set_error_group_callback`, you will start your code search at `lib/new_relic/agent.rb` looking for a `set_error_group_callback` method.

In any event, I'd like to see any refactoring of this class discussed and then tackled with a dedicated separate story.